cmd/acr/purge.go

```diff
@@ -59,15 +59,22 @@ var (
 	concurrencyDescription = fmt.Sprintf("Number of concurrent purge tasks. Range: [1 - %d]", maxPoolSize)
 )
 
+// Default settings for regexp2
+const (
+	defaultRegexpOptions             regexp2.RegexOptions = regexp2.RE2 // This option will turn on compatibility mode so that it uses the group rules in regexp
+	defaultRegexpMatchTimeoutSeconds uint64               = 1800
+)
+
 // purgeParameters defines the parameters that the purge command uses (including the registry name, username and password).
 type purgeParameters struct {
 	*rootParameters
-	ago         string
-	keep        int
-	filters     []string
-	untagged    bool
-	dryRun      bool
-	concurrency int
+	ago           string
+	keep          int
+	filters       []string
+	filterTimeout uint64
```

Comment:
The timeout is because with lookahead regex there is a possibility the state machine ends up in an infinite loop. I wasn't really sure what a reasonable default timeout should be so I picked something I thought was reasonable and exposed this flag in case it isn't reasonable. I'm not sure if there are legitimate state-machines that can take over 1 minute or not so I'm just trying to be cautious.

Reply:
OK. Is default 1800 seconds too long? Do you have a rough estimation of the normal case? For example, the original user case that we are trying to address in the case. Maybe default 60 seconds?